frontend/src/app/main/data/workspace/selection.cljs

```diff
@@ -495,18 +496,29 @@
 
                   id-original     (first selected)
 
-                  selected        (->> changes
+                  new-selected    (->> changes
                                        :redo-changes
                                        (filter #(= (:type %) :add-obj))
                                        (filter #(selected (:old-id %)))
                                        (map #(get-in % [:obj :id]))
                                        (into (d/ordered-set)))
 
-                  id-duplicated   (first selected)]
-               ;; Warning: This order is important for the focus mode.
-              (rx/of (dch/commit-changes changes)
-                     (select-shapes selected)
-                     (memorize-duplicated id-original id-duplicated)))))))))
+                  dup-frames      (->> changes
+                                       :redo-changes
+                                       (filter #(= (:type %) :add-obj))
+                                       (filter #(selected (:old-id %)))
+                                       (filter #(= :frame (get-in % [:obj :type])))
+                                       (map #(vector (:old-id %) (get-in % [:obj :id]))))
+
+                  id-duplicated   (first new-selected)]
+
+              ;; Warning: This order is important for the focus mode.
+              (rx/merge
```

Comment:
If order is important, I think `rx/concat` should be used instead of merge

Reply:
The comment was missplaced. Anyway, I think concat is better here.